src/main/java/edu/slu/action/ObjectAction.java

```diff
@@ -1252,24 +1252,37 @@ public boolean checkModified(HttpServletRequest req, JSONObject jo){
                     return true;
                 }
             }
+            else if(jo.has("__deleted")){
+                lastModifiedDateObj = jo.getJSONObject("__deleted").getString("time");
+            }
             else{
                 //If it's not from RERUM, I have no way to check.  Consider it modified.
                 //System.out.println("Obj does not have __rerum, consider it modified");
                 return true;
             }
-            //Note that dates like 2021-05-26T10:39:19.328 have been rounded to 2021-05-26T10:39:19.328 in browser headers.  Account for that here.
-            if(lastModifiedDateObj.contains(".")){
-                //If-Modified-Since and Last-Modified headers are rounded.  Wed, 26 May 2021 10:39:19.629 GMT becomes Wed, 26 May 2021 10:39:19 GMT.
-                lastModifiedDateObj = lastModifiedDateObj.split("\\.")[0];
+            try{
+                //Note that dates like 2021-05-26T10:39:19.328 have been rounded to 2021-05-26T10:39:19.328 in browser headers.  Account for that here.
+                if(lastModifiedDateObj.contains(".")){
```

Comment:
This if is probably not needed, since the result is the same as just running the next line.